bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/values/XMLText.java

```diff
@@ -142,6 +142,9 @@ public int hashCode() {
 
     @Override
     public Type getType() {
+        if (this.data.isEmpty()) {
```

Comment:
Since an XML text value is inherently immutable, can't we check if `data` is empty, and set `xml<never>` as the type in the constructor of `XMLText` itself?

Reply:
Checks like :  
```
string empty = "";
'xml:Text a = xml `${empty}`;
any y = a;
assertEquality(y is xml<never>, true);
```
 fail if it is put in the constructor. The values seems to be getting resolved after the constructor is called.

Reply:
Wouldn't adding 

`this.type = data.isEmpty() ? PredefinedTypes.TYPE_XML_NEVER : PredefinedTypes.TYPE_TEXT;` 

to the constructor and updating `getType` as

```java
    @Override
    public Type getType() {
        return this.type;
    }
```

work?